tests/cover/test_float_nastiness.py

```diff
@@ -150,3 +152,31 @@ def test_very_narrow_interval():
     def test(f):
         assert lower_bound <= f <= upper_bound
     test()
+
```

Comment:
Since you're fixing a bug (that we produced `floats()` examples outside the correct interval), I'm inclined to suggest a test which:

- specifically checks for that 
- would have failed before

Does that make sense?

If you keep the deprecation warning for the "no floats" case, that can be exempt.

Reply:
I know you have `test_float_free_interval_is_invalid` checking a particular case where we would give an example that's not in the interval. I guess I'm just saying maybe some more general check along those lines, intended for the non-deprecation-warning bug case that you fixed, would be good.

Reply:
Done.